drive_option.go

```diff
@@ -0,0 +1,33 @@
+package onedrive
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// DriveOption defines a search criteria scoped
+// to either onedrive or sharepoint
+type DriveOption struct {
+	Query  string
+	Scope  DriveScope
+	SiteID string
```

Comment:
I'm a little confused as to why `Scope` and `SiteID` are separate - it seems like `SiteID` is only applicable when `Scope` is set to a specific value? It seems like an opportunity for some polymorphism to hide details from the caller?

How about if we made `DriveScope` an interface that exposes `PathPrefix()`. When the scope is a sharepoint site, we store the SiteID in a struct that implements that interface?

eg.

```go
type DriveScope interface {
  PathPrefix() string
}

type MeScope struct {}
func (m MeScope) PathPrefix string {
	return "/me"
}

type SharepointSiteScope struct {
	SiteID string
}
func (s SharepointSiteScope) PathPrefix() string {
	return "/sites/"+s.SiteID
}
```

What do you think?

Reply:
Oh yeah thats a good call actually and I like that better I'll update 👍

Reply:
@BRMatt done